MoreLinq/ListLike.cs

```diff
@@ -21,34 +21,35 @@ namespace MoreLinq
     using System.Collections.Generic;
 
     /// <summary>
-    /// Represents an list-like (indexable) data structure.
+    /// Represents a union over list types implementing either
+    /// <see cref="IList{T}"/> or <see cref="IReadOnlyList{T}"/>, allowing
+    /// both to be treated the same.
     /// </summary>
 
-    interface IListLike<out T>
+    struct ListLike<T> // TODO Can be readonly when using C# 7.2
     {
-        int Count { get; }
-        T this[int index] { get; }
-    }
+        readonly IList<T> _l;
+        readonly IReadOnlyList<T> _r;
```

Comment:
These member names are quite cryptic.

Reply:
Tried to address this with 064fb6151586c2e467920662678c1dc8e0231f47. Perhaps `rw` and `rx` are still cryptic, but wanted to keep them the same length (trying to hard?) so these are *nix inspired (`rw` = read-write, `rx` = read-only).